src/k2/cpo/service/CPOService.cpp

```diff
@@ -597,9 +607,41 @@ CPOService::handlePersistenceClusterGet(dto::PersistenceClusterGetRequest&& requ
         return RPCResponse(Statuses::S500_Internal_Server_Error("unable to read persistence cluster data"), dto::PersistenceClusterGetResponse());
     };
 
-    K2LOG_I(log::cposvr, "Found persistence cluster in: {}", cpath);
+    K2LOG_D(log::cposvr, "Found persistence cluster in: {}", cpath);
     dto::PersistenceClusterGetResponse response{.cluster=std::move(persistenceCluster)};
     return RPCResponse(Statuses::S200_OK("persistence cluster found"), std::move(response));
 }
 
+seastar::future<std::tuple<Status, dto::MetadataPersistResponse>>
+CPOService::handleMetadataPersist(dto::MetadataPersistRequest&& request){
+    K2LOG_D(log::cposvr, "Received metadata persist request for partition {}", request.partitionName);
+    auto records = _metadataRecords.find(request.partitionName);
+    if (records == _metadataRecords.end()) {
+        std::vector<dto::MetadataRecord> log;
+        dto::MetadataRecord element{.plogId=std::move(request.new_plogId), .sealed_offset=0};
+        log.push_back(std::move(element));
+        _metadataRecords[std::move(request.partitionName)] = std::move(log);
+    }
+    else{
+        records->second.back().sealed_offset = request.sealed_offset;
+        dto::MetadataRecord element{.plogId=std::move(request.new_plogId), .sealed_offset=0};
+        records->second.push_back(std::move(element));
+    }
+    
+    return RPCResponse(Statuses::S200_OK("metadata log updates successfully"), dto::MetadataPersistResponse());
+}
+
+seastar::future<std::tuple<Status, dto::MetadataGetResponse>>
+CPOService::handleMetadataGet(dto::MetadataGetRequest&& request){
+    K2LOG_D(log::cposvr, "Received metadata get request for partition {}", request.partitionName);
+    auto records = _metadataRecords.find(request.partitionName);
+    if (records == _metadataRecords.end()) {
+        return RPCResponse(Statuses::S409_Conflict("request parition's metadata does not exist"), dto::MetadataGetResponse());
+    }
+
+    dto::MetadataGetResponse response{.records=records->second};
+    return RPCResponse(Statuses::S200_OK("metadata log updates successfully"), std::move(response));
```

Comment:
The message is wrong here. Should be probably empty

Reply:
fixed